scipy/signal/_peak_finding.py

```diff
@@ -233,6 +236,733 @@ def argrelextrema(data, comparator, axis=0, order=1, mode='clip'):
+    peak_widths
+        Directly calculate the width of peaks.
+
+    Notes
+    -----
+    Because this function searches for local maxima by direct sample comparison,
+    the determined peak locations can be off for noisy signals if the noise
+    changes the position of a local maximum. In those cases, smoothing the signal
+    may be more appropriate before searching for peaks or using other peak
+    finding and fitting methods (like `find_peaks_cwt`).
+
+    Some additional comments on filtering with requirements:
+
+    * Almost all filter options (excluding filtering by distance) allow the
+      specification of half-open or closed intervals, e.g ``1`` or ``(1, None)``
+      define the half-open interval :math:`[3, \\infty]` while ``(None, 1)``
+      defines the interval :math:`[-\\infty, 3]`.
+    * The border is always included in the interval used to select valid peaks.
+    * The order of input arguments for the different filters mirrors the actual
+      order in which they are applied. In most cases this order is the fastest
+      one because faster filter operations are applied first to reduce the
+      number of peaks that need to be evaluated later.
+    * Filtering by distance is accomplished by iterating over all peaks in
+      descending order based on their height and removing all lower peaks that
+      are to close. This option can be quite slow if a large number of peaks
+      needs to be evaluated. Try to reduce the number of peaks with cheaper
+      (previous) filter options first.
+    * Use `wlen` to reduce the time it takes to filter by prominence or width if
+      `x` is large or has many local maxima (see `peak_prominences`).
+    * For several filter options the interval borders can be specified with
+      arrays matching `x` in shape which enables dynamic filter constrains.
+
+    .. versionadded:: 1.1.0
+
+    Examples
+    --------
+    Create test signal and find peaks with a minimal prominence of 0.5
+
+    >>> from scipy.signal import find_peaks
+    >>> t = np.linspace(0, 8 * np.pi, 1000)
+    >>> x = np.sin(t) + 0.2 * np.sin(10 * t) + 0.1 * t
+    >>> peaks, properties = find_peaks(x, prominence=0.5)
+    >>> peaks
+    array([ 57, 306, 556, 806])
+
+    Plot results
+
+    >>> import matplotlib.pyplot as plt
+    >>> countour_height = x[peaks] - properties['prominences']
+    >>> plt.plot(t, x)
+    >>> plt.plot(t[peaks], x[peaks], "x")
+    >>> plt.vlines(t[peaks], ymin=countour_height, ymax=x[peaks])
+    """
+    # _argmaxima1d expects array of dtype 'float64'
+    x = np.asarray(x, dtype=np.float64)
+    if x.ndim != 1:
+        raise ValueError('`x` must have exactly one dimension')
+    if distance is not None and distance < 1:
+        raise ValueError('`distance` must be greater or equal to 1')
+
+    peaks = _argmaxima1d(x)
+    properties = {}
+
+    if peaks.size == 0:
+        # Skip filtering if no peaks where found
+        return peaks, properties
```

Comment:
Do you need to return early? In theory the code below could be made to handle the `peaks.size == 0` case if they don't already. Doing so will presumably have negligible processing overhead. Then my complaint about the `properties` dict will automatically be taken care of by the code that follows, which will set the dict key/value pairs properly.

Reply:
You are right, this case is already handled by the rest of the code. I'll remove this if-statement.